backend/drm/renderer.c

```diff
@@ -277,22 +263,38 @@ bool drm_plane_init_surface(struct wlr_drm_plane *plane,
 		drm_format = wlr_drm_format_dup(&format_no_modifiers);
 	}
 
+	struct wlr_drm_format *drm_format_linear =
+		calloc(1, sizeof(struct wlr_drm_format) + sizeof(uint64_t));
+	if (drm_format_linear == NULL) {
+		return false;
```

Comment:
Don't we need to free `drm_format`?

Reply:
I also don't like we are doing an allocation we might not need but the alternative seems much more complex code ...

Reply:
Good catch, fixed.

> I also don't like we are doing an allocation we might not need but the alternative seems much more complex code ...

Yes. It shouldn't matter in practice, it's not a hot path at all, and micro-optimizations detrimental to code readability are a bad idea.